### DISCUSSION TITLE
How can I include timestamps/loglevel/callers in the structure while in Development mode?

### DISCUSSION THREAD

**Discussion by mirumirumi** [uber-go/zap#1175] [Q&A]
I prefer all conditions generated by `NewDevelopment()`.
However, I just don't like the fact that `timestamp`, `loglevel` and `caller` are outside the structured log.

![image](https://user-images.githubusercontent.com/88871226/189148930-80f98b15-8a2b-471d-b608-27fff88a50b2.png)

Using `NewProduction()` solves this problem, but then other problems occur, such as no debug level logs.

How can I include it in the structured log as NewDevelopment?

**Comment by abhinav** [comment]
Hello, @mirumirumi. Thanks for your question.
You can build a completely custom configuration,
or you can take the logger configuration of NewDevelopment,
and use the encoder configuration of NewProduction.

```go
cfg := zap.NewDevelopmentConfig()
cfg.EncoderConfig = zap.NewProductionEncoderConfig()
logger, err := cfg.Build()
```
